### DISCUSSION TITLE
Is it possible to use std::chrono::system_clock::now()?

### DISCUSSION THREAD

**Discussion by Riksu9000** [InfiniTimeOrg/InfiniTime#1656] [Q&A]
On the PineTime, `std::chrono::system_clock::now().time_since_epoch()` returns the same value every time. In InfiniSim it seems to work correctly. Do we have to configure something to get it to work?

[sysclock.patch.zip](https://github.com/InfiniTimeOrg/InfiniTime/files/10848773/sysclock.patch.zip)

**Comment by kestara** [comment]
`std::chrono::system_clock` is not supported in InfiniTime hardware.

As per comment [here](https://github.com/InfiniTimeOrg/InfiniTime/discussions/1582#discussioncomment-4949020), the alternative to use is `DateTimeController`

> Instead, you should try using the DateTimeController (https://github.com/InfiniTimeOrg/InfiniTime/blob/develop/src/components/datetime/DateTimeController.h) to get the date and time of the PineTime. You can take some inspirations from the existing apps and watchfaces (ex : https://github.com/InfiniTimeOrg/InfiniTime/blob/develop/src/displayapp/screens/WatchFaceDigital.cpp#L94).

**Reply by Riksu9000** [reply]
Oops, should've looked for other discussions. Still I'd be interested to know what it would take to get std::chrono::system_clock::now working.

**Comment by JF002** [comment]
To be honest, I don't know what needs to be done to implement a `system_clock' on bare metal. I remember I searched a bit when I implemented that part of the project, but I couldn't find much info about this.

I guess we need to implement some low-level function to "link" std::chrono::system_clock to the actual hardware timer. If someone has more info about this, I'm also interested!

**Reply by lmamane** [reply]
> I don't know what needs to be done to implement a `system_clock' on bare metal.

> I guess we need to implement some low-level function to "link" std::chrono::system_clock to the actual hardware timer.

Implemented in pull request #2162
